enterprise/templates/enterprise/grant_data_sharing_permissions.html

```diff
@@ -43,10 +47,20 @@
                             data-warning-message="${ messages['warning'] }"
                         />
                         <label for="register-data_sharing_consent">
-                            ${_("I agree to allow {platform_name} to share data about my enrollment, completion and performance in all {platform_name} courses and programs where my enrollment is sponsored by {provider}.*").format(provider=sso_provider, platform_name=platform_name)}
+                            % if not course_id:
```

Comment:
Same here -- we should have a dedicated switch instead of hanging on `course_id`